warehouse/query-core/src/main/java/datawave/query/attributes/Content.java

```diff
@@ -117,13 +145,31 @@ public void write(Kryo kryo, Output output, Boolean reducedResponse) {
         super.writeMetadata(kryo, output, reducedResponse);
         output.writeString(this.content);
         output.writeBoolean(this.toKeep);
+        output.writeBoolean(this.source != null);
+        if (source != null) {
+            output.writeString(this.source.getClass().getCanonicalName());
+            source.write(kryo, output, reducedResponse);
+        }
     }
 
     @Override
     public void read(Kryo kryo, Input input) {
         super.readMetadata(kryo, input);
         this.content = input.readString();
         this.toKeep = input.readBoolean();
+        boolean hasSource = input.readBoolean();
+        if (hasSource) {
+            String clazz = input.readString();
+            Class sourceClass;
+            try {
+                sourceClass = Class.forName(clazz);
+                source = (Attribute<?>) sourceClass.newInstance();
+            } catch (ClassNotFoundException | InstantiationException | IllegalAccessException e) {
+                throw new RuntimeException("could not parse source", e);
+            }
+
+            source.read(kryo, input);
+        }
```

Comment:
This looks good. Can you update the ContentTest to include variants with a source attribute?